src/matcher_combinators/midje.clj

```diff
@@ -25,6 +25,29 @@
       (checking/as-data-laden-falsehood
         {:notes [(str "Input wasn't a matcher: " matcher)]}))))
 
+(defn var->qualified-ref [datatype]
+  (symbol (str
+            (-> parser/type-map datatype meta :ns ns-name)
+            "/"
+            (-> parser/type-map datatype meta :name))))
+
+(defmacro match-with [type->matcher expected-matcher]
```

Comment:
For context, `type->matcher` should be a map looking something like https://github.com/nubank/matcher-combinators/pull/29/files#diff-51dee594687ff558df6b1fc75a4cc561R35 (though in this case should point to the actual function and not a var pointing the function).

This will be clarified as I wrap up the PR/add documentation